src/ucp/rma/amo_sw.c

```diff
@@ -405,7 +405,7 @@ static ucs_status_t ucp_proto_amo_sw_progress_post(uct_pending_req_t *self)
 static ucs_status_t
 ucp_proto_amo_sw_init_post(const ucp_proto_init_params_t *init_params)
 {
-    UCP_RMA_PROTO_INIT_CHECK(init_params, UCP_OP_ID_AMO_POST);
+    UCP_RMA_PROTO_INIT_CHECK(init_params, UCP_OP_ID_AMO_POST, true);
```

Comment:
looks like need UCP_AMO_PROTO_INIT_CHECK to check DT is contig
and remove contig check from UCP_RMA_PROTO_INIT_CHECK

Reply:
It introduces a naming problem, because it would be used in `rndv` as well.

Then we'll have `rndv` calling `UCP_AMO_PROTO_INIT_CHECK` ?

Reply:
IMO whoever is calling UCP_RMA_PROTO_INIT_CHECK with false - should just check op_id directly - one condition check should not deserve a macro
and the macro which checks op_id and CONTIG - can call it UCP_PROTO_INIT_CHECK_CONTIG and put into proto_common.h